server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2211,6 +2214,31 @@ else if (details != null && !details.isEmpty()) {
         return _tmpltDao.findById(id);
     }
 
+    void validateDetails(VMTemplateVO template, Map<String, String> details) {
+        if (details == null || details.isEmpty()) {
+            return;
+        }
+        String bootMode = details.get(ApiConstants.BootType.UEFI.toString());
+        if (bootMode == null) {
+            return;
+        }
+        if (template.isDeployAsIs()) {
+            String msg = "Deploy-as-is templates can not have UEFI parameter set. Settings are read directly from the template";
+            throw new InvalidParameterValueException(msg);
+        }
+        try {
+            String mode = bootMode.trim().toUpperCase();
+            ApiConstants.BootMode.valueOf(mode);
+            details.put(ApiConstants.BootType.UEFI.toString(), mode);
+            return;
+        } catch (IllegalArgumentException e) {
+            String msg = String.format("Invalid %s: %s specified. Valid values are: %s",
+                    ApiConstants.BOOT_MODE, bootMode, Arrays.toString(ApiConstants.BootMode.values()));
+            s_logger.error(msg);
```

Comment:
We could pass the exception as parameter of  `s_logger.error`.

Reply:
The exception is not thrown but logged instead. So explicitly throwing it in the following line